### ISSUE TITLE
member setter: Type of "x" could not be determined because it refers to itself (reportGeneralTypeIssues)

### ISSUE THREAD

**Issue by nschloe** [microsoft/pyright#1220]
The following is a recommended way of implementing a member setter in a Python class (see [here](https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters)). However, pyright yields an error:
```python
class C:
    def __init__(self):
        self._x = None

    @property
    def x(self):
        print("getter of x called")
        return self._x

    @x.setter
    def x(self, value):
        print("setter of x called")
        self._x = value


c = C()
c.x = "foo"  # setter called
```
```
  10:6 - error: Type of "x" could not be determined because it refers to itself (reportGeneralTypeIssues)
```
Is this a false positive or is there another way of implementing a setter?

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. You can work around it in the meantime by providing a type annotation for `self._x` or for the return type of the `x` property getter.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.91, which I just published. It will also be included in the next release of pylance.
